angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete.component.spec.ts

```diff
@@ -24,12 +24,44 @@ import { SprkInputDirective } from '../../directives/inputs/sprk-input/sprk-inpu
 })
 class WrappedAutocompleteComponent {}
 
+@Component({
+  selector: 'sprk-dynamic-test',
+  template: `
+    <button class="changeButton" (click)="changeMe()">Change Content</button>
+
+    <sprk-autocomplete isOpen="true" itemSelectedEvent="itemSelectedEvent">
+      <div sprkAutocompleteInputContainer>
+        <input sprkInput />
+      </div>
+      <ul sprkAutocompleteResults>
+        <li
+          sprkAutocompleteResult
+          *ngFor="let item of list; let i = index"
+          id="{{ item }}"
+        ></li>
+      </ul>
+    </sprk-autocomplete>
+  `,
+})
+class DynamicComponent {
+  public itemSelectedEvent;
+  public list = ['item1', 'item2', 'item3', 'item4'];
+  public newList = ['item1', 'item2', 'item3', 'item4', 'new_id'];
+  changeMe(): void {
+    this.list = this.newList;
+  }
+}
```

Comment:
This test component uses a button (`changeMe()`) to change the data used to populate the control. This triggers the `.changes` logic in Autocomplete which should rebind the click events to the new items.

Reply:
this is nice because it also tests if someone affects the autocomplete in some other way other than typing

Reply:
agreed!